ResearchKit/ActiveTasks/ORKPSATStepViewController.m

```diff
@@ -0,0 +1,246 @@
+ are permitted provided that the following conditions are met:
+ 
+ 1.  Redistributions of source code must retain the above copyright notice, this
+ list of conditions and the following disclaimer.
+ 
+ 2.  Redistributions in binary form must reproduce the above copyright notice,
+ this list of conditions and the following disclaimer in the documentation and/or
+ other materials provided with the distribution.
+ 
+ 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+ may be used to endorse or promote products derived from this software without
+ specific prior written permission. No license is granted to the trademarks of
+ the copyright holders even if such marks are included in this software.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#import "ORKPSATStepViewController.h"
+#import "ORKActiveStepViewController_Internal.h"
+#import "ORKStepViewController_Internal.h"
+#import "ORKPSATContentView.h"
+#import "ORKPSATStep.h"
+#import "ORKVerticalContainerView.h"
+#import "ORKActiveStepView.h"
+#import "ORKPSATKeyboardView.h"
+
+
+@interface ORKPSATStepViewController () <ORKPSATKeyboardViewDelegate>
+
+@property (nonatomic, strong) NSMutableArray *samples;
+@property (nonatomic, strong) ORKPSATContentView *psatContentView;
+@property (nonatomic, strong) NSArray *digits;
+@property (nonatomic, assign) NSUInteger currentDigitIndex;
+@property (nonatomic, assign) NSInteger currentAnswer;
+@property (nonatomic, strong) ORKActiveStepTimer *timeoutTimer;
+@property (nonatomic, assign) NSTimeInterval answerStart;
+@property (nonatomic, assign) NSTimeInterval answerEnd;
+
+@end
+
+@implementation ORKPSATStepViewController
+
+- (instancetype)initWithStep:(ORKStep *)step {
+    self = [super initWithStep:step];
+    
+    if (self) {
+        self.suspendIfInactive = YES;
+    }
+    
+    return self;
+}
+
+- (ORKPSATStep *)psatStep {
+    return (ORKPSATStep *)self.step;
+}
+
+- (NSArray *)arrayWithPSATDigits {
+    NSMutableArray *array = [[NSMutableArray alloc] initWithCapacity:[self psatStep].seriesLength + 1];
+    NSUInteger digit = 0;
+    for (NSUInteger i = 0; i < [self psatStep].seriesLength + 1; i++) {
+        do
+        {
+            digit = (arc4random() % (9)) + 1;
```

Comment:
I think the sequence is often fixed because the technician has to fill in a pre-print sheet of paper with the results. So it'd be mainly a "physical" constraint.
I didn't see any data than could indicate scientists need the same sequence to compare results amongst cohorts.

Moreover, contrary to actual test that are conducted from time to time, this task could be used more often by a patient, and some papers mentioned it could exist a learning curve that could artificially improve the result, if the sequence is fixed between tests.

So I think at this stage, without more scientific evidence, there's no need to add support for a fixed sequence.